### PR TITLE
ALL_MIX TeamType 

### PR DESCRIPTION
**Pull request by Jmiceli17**
Added ALL_MIX TeamType to support true random selection of teammates for FCP training

### CODE REVIEW

File: oai_agents/common/tags.py [hunk 1]
```diff
@@ -60,7 +67,8 @@ def map_to_index(teamtype):
             TeamType.SELF_PLAY: 9,
             TeamType.SELF_PLAY_LOW: 10,
             TeamType.SELF_PLAY_MEDIUM: 11,
-            TeamType.SELF_PLAY_HIGH: 12
+            TeamType.SELF_PLAY_HIGH: 12,
+            TeamType.ALL_MIX: 13    # TODO: Should this go here or after HLR?
```
Comment (Jmiceli17) [inline oai_agents/common/tags.py]: Is this in the right spot for the `map_to_index`?

File: scripts/train_agents.py [hunk 2]
```diff
@@ -83,25 +83,11 @@ def SP_w_SP_Types(args,
 
 
 def FCP(args, pop_force_training, fcp_force_training, parallel):
```
Comment (avv-va) [inline scripts/train_agents.py]: So there are two types of FCP, one is the traditional FCP that uses random teammates (i.e. ALL_MIX), one is our own version that uses certain types HIGH_FIRST, MEDIUM_FIRST, etc. The reason we have our version is that when we used the traditional FCP it got ~0 reward and then we decided to add all these different types for teammates_collection.

I suggest creating two functions: `FCP_traditional` that uses `ALL_MIX`, `FCP_mhri` that uses the default setting we have right now in the main branch.

File: oai_agents/common/tags.py (hunk 1 above)
Comment (avv-va) [reply oai_agents/common/tags.py]: yes

**Review by avv-va — COMMENTED** [review]
This looks pretty good! Nice job!

File: scripts/train_agents.py (hunk 2 above)
Comment (Jmiceli17) [reply scripts/train_agents.py]: added!

**Review by narendasan — APPROVED** [review]
LGTM!

**Pull request merged**